libclc/generic/lib/integer/add_sat.cl

```diff
@@ -13,12 +13,20 @@ _CLC_DECL ulong  __clc_add_sat_u64(ulong, ulong);
 
 _CLC_OVERLOAD _CLC_DEF char add_sat(char x, char y) {
   short r = x + y;
-  return convert_char_sat(r);
```

Comment:
1. Why are only char and uchar cases processed?
2. What are the specific problems in the implementation of convert_*_sat function?

Reply:
do not check the impementation of convert_char_sat(short)

Reply:
do not know why only uchar implementation has error, but it works when I changed the function

Reply:
It is unreasonable to deal with one of the same types of problems without knowing the underlying cause.